TypeCobol.LanguageServer/TypeCobolServer.cs

```diff
@@ -984,4 +1020,60 @@ public CodeElementWrapper(CodeElement codeElement)
 
         public List<Token> ArrangedConsumedTokens { get; set; }
     }
+
+    internal static class ToolTipHelper
+    {
+        /// <summary>
+        /// Gives self and children text lines of a node, each text line is associated
+        /// with its depth.
+        /// </summary>
+        /// <param name="node">Node to explore.</param>
+        /// <param name="level">Starting level, default is 0.</param>
+        /// <returns>IEnumerable of Tuple, Item1 is depth, Item2 is ITextLine.</returns>
+        private static IEnumerable<Tuple<int, ITextLine>> GetLinesWithLevel(Node node, int level = 0)
+        {
+            if (node == null) yield break;
+
+            // Lines of current level in node
+            if (node.CodeElement?.ConsumedTokens != null)
+            {
+                var lines = new List<ITextLine>();
```

Comment:
Why a list here ?
It seems you only need to remember the last line seen.

Reply:
Yes, you're right. This has been fixed with a47405251aca00c87ab802227c12827430bb7377.